articles/quickstart/native/ios-swift/09-sign-in-with-apple.md

```diff
@@ -0,0 +1,197 @@
+  let authorizationButton = ASAuthorizationAppleIDButton()
+
+  // Add Callback on Touch
+  authorizationButton.addTarget(self, action: #selector(handleAuthorizationAppleIDButtonPress), for: .touchUpInside)
+
+  //Add button to the UIStackView
+  self.loginProviderStackView.addArrangedSubview(authorizationButton)
+}
+```
+
+Call this function in `viewDidLoad`:
+
+```swift
+override func viewDidLoad() {
+    super.viewDidLoad()
+
+    setupProviderLoginView()
+}
+```
+
+## Handle Authorization
+
+The following code shows the implementation of a button handler that initializes an authorization request and launches
+the native SIWA flow:
+
+```swift
+@objc
+func handleAuthorizationAppleIDButtonPress() {
+    // Create the authorization request
+    let request = ASAuthorizationAppleIDProvider().createRequest()
+
+    // Set scopes
+    request.requestedScopes = [.email, .fullName]
+
+    // Setup a controller to display the authorization flow
+    let controller = ASAuthorizationController(authorizationRequests: [request])
+
+    // Set delegates to handle the flow response.
+    controller.delegate = self
+    controller.presentationContextProvider = self
+
+    // Action
+    controller.performRequests()
+}
+```
+
+:::note
+At this time the only supported scopes are `email` and `fullname`. For more information, see [ASAuthorization.scope](https://developer.apple.com/documentation/authenticationservices/asauthorization/scope).
+:::
+
+At this point, you will have a few compile errors to deal with. Before moving forward with all the code, let's remove those errors by providing some basic delegates.
+
+### Add authorization extensions
+
+Add the following to the end of the `ViewController.swift` file. This enables the authorization flow UI to be displayed from our controller:
+
+```swift
+extension ViewController: ASAuthorizationControllerPresentationContextProviding {
+    func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
+        return self.view.window!
+    }
+}
+```
+
+Next, add a stub delegate for the `ASAuthorizationController` to the end of the `ViewController.swift` file. This code will handle the response sent to the `ASAuthorizationController`. You will come back to this one later.
+
+```swift
+extension ViewController: ASAuthorizationControllerDelegate {
+}
+```
+
+### Add the Sign In With Apple capability
+
+To enable Sign In With Apple in your application, you need to enable this capability to your app, as shown:
+
+![Add capabilities](/media/articles/ios/swift/add-capability.png)
+
+:::panel Checkpoint
+Start the application and click the **Sign In With Apple** button. You should see the default sign-in dialog appear:
+
+![Default sign-in dialog](/media/articles/ios/swift/authz-dialog.png)
+:::
+
+## Process the Authorization Response
+
+Add the following to the end of your `ViewController.swift` file.
+
+```swift
+extension ViewController: ASAuthorizationControllerDelegate {
+
```

Comment:
Extra blank lines here after opening functions - style is different to other code blocks.